### PR TITLE
ci(helm-test): add upgrade check

### LINKED ISSUE #276: TRG 5.11 - Upgradeability
Hi @tom-rm-meyer-ISST,

I can't find action to test upgradeability in the helm test workflow. I understand it's first release and there is no room to test that however recommend to have it in place for future usage.

More details: https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-11
Updates https://github.com/eclipse-tractusx/sig-release/issues/510

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description

closes #276 

Note: in [TRG 5.09](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-09) it's indicated that we use first tractusx-dev[CHART] -> I expected it to be a typo and used charts/[CHART]

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: .github/workflows/helm-test.yml [hunk 1]
```diff
@@ -105,3 +111,14 @@ jobs:
                     helm repo add postgresql https://charts.bitnami.com/bitnami
                     ct install --charts charts/puris --helm-extra-set-args "--set backend.image.repository=${{ env.REGISTRY }}/${{ env.APP_BACKEND_NAME}} --set backend.image.tag=${{ env.TAG }} --set frontend.image.repository=${{ env.REGISTRY }}/${{ env.APP_FRONTEND_NAME }} --set frontend.image.tag=${{ env.TAG }}"
                 if: github.event_name != 'pull_request' || steps.list-changed.outputs.changed == 'true'
+
+            # Upgrade the released chart version with the locally available chart
+            # default value for event_name != workflow_dispatch
+            -   name: Run helm upgrade
+                run: |
+                    helm repo add bitnami https://charts.bitnami.com/bitnami
+                    helm repo add postgresql https://charts.bitnami.com/bitnami
```
Comment (tomaszbarwicki) [inline .github/workflows/helm-test.yml:120]: Not sure if it makes sense to add same repo under different name.
Following TRG, suggest to add our helm repo, replace with:
```suggestion
                    helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev
```

File: .github/workflows/helm-test.yml [hunk 2]
```diff
@@ -105,3 +111,14 @@ jobs:
...
+                    helm install puris charts/puris --version ${{ github.event.inputs.upgrade_from || '1.0.0' }}
```
Comment (tomaszbarwicki) [inline .github/workflows/helm-test.yml:121]: I think that won't work if you have newer version, recommend to install it from our helm repo:

```suggestion
                    helm install puris tractusx-dev/puris --version ${{ github.event.inputs.upgrade_from || '1.0.0' }}
```

File: .github/workflows/helm-test.yml [hunk 3]
```diff
@@ -105,3 +111,14 @@ jobs:
...
+                    helm dependency update charts/puris
+                    helm upgrade [NAME] charts/puris
```
Comment (tomaszbarwicki) [inline .github/workflows/helm-test.yml:123]: Replace [NAME] placeholder. 

```suggestion
                    helm upgrade puris charts/puris
```

File: .github/workflows/helm-test.yml [hunk 4]
```diff
@@ -30,6 +30,12 @@ on:
                 default: 'kindest/node:v1.27.3'
                 required: false
                 type: string
+            upgrade_from:
+                description: 'chart version to upgrade from'
+                # chart version from 3.1 release as default
```
Comment (tomaszbarwicki) [inline .github/workflows/helm-test.yml:35]:
```suggestion
                # chart version from 1.0.0 release as default
```

**Review by tomaszbarwicki — CHANGES_REQUESTED** [review]

File: .github/workflows/helm-test.yml (hunk 1 above)
Comment (tom-rm-meyer-ISST) [reply .github/workflows/helm-test.yml:120]: Ah missed this detail

**Comment by tom-rm-meyer-ISST** [comment]
incorporated changes, please review

**Review by tomaszbarwicki — APPROVED** [review]
LGTM

**Pull request merged**